pymc3/tests/test_smc.py

```diff
@@ -47,31 +45,25 @@ def two_gaussians(x):
             X = pm.Uniform('X', lower=-2, upper=2., shape=n)
             llk = pm.Potential('muh', two_gaussians(X))
 
-        self.step = smc.SMC(
-            n_chains=self.n_chains,
-            tune_interval=self.tune_interval,
-            model=self.ATMIP_test)
-
         self.muref = mu1
 
     @pytest.mark.parametrize(['n_jobs', 'stage'], [[1, 0], [2, 6]])
     def test_sample_n_core(self, n_jobs, stage):
 
-        mtrace = smc.sample_smc(
-            n_steps=self.n_steps,
-            step=self.step,
-            stage=stage,
-            n_jobs=n_jobs,
-            progressbar=True,
-            homepath=self.test_folder,
-            model=self.ATMIP_test,
-            rm_flag=True)
+        mtrace = smc.sample_smc(samples=self.samples,
+                                n_chains=self.n_chains,
+                                stage=stage,
+                                n_jobs=n_jobs,
+                                progressbar=True,
+                                homepath=self.test_folder,
+                                model=self.ATMIP_test,
+                                rm_flag=True)
 
         x = mtrace.get_values('X')
         mu1d = np.abs(x).mean(axis=0)
         np.testing.assert_allclose(self.muref, mu1d, rtol=0., atol=0.03)
         # Scenario IV Ching, J. & Chen, Y. 2007
-        assert np.round(np.log(self.ATMIP_test.marginal_likelihood)) == -11.0
+        assert np.round(np.log(self.ATMIP_test.marginal_likelihood)) == -12.0
```

Comment:
Is this marginal now what you expected? Ergo did it improve it?

Reply:
The _marginal likelihoods_ are still off for models where the prior is _too different_ from the likelihood. I have been testing this on the beta-binomial model, well in fact I do not test the _marginal likelihoods_ directly, but the Bayes Factors.

My primary motivation for all these changes was to improve the _marginal likelihoods_, but it seems to be more difficult than I thought, or I am making a big _obvious_ mistake at some point :)

Reply:
That ESS is mentioned in various papers, for example 
[Sequential Monte Carlo for static Bayesian models with independent MCMC proposals](https://eprints.qut.edu.au/101729/) and [ Effective Sample Size for Importance Sampling based on discrepancy measures](https://arxiv.org/abs/1602.03572), this last paper compare several ESS. I still need to read it in detail.